OpenRA.Mods.Common/Scripting/Properties/CombatProperties.cs

```diff
@@ -90,13 +90,19 @@ public GeneralCombatProperties(ScriptContext context, Actor self)
 		public void Attack(Actor targetActor, bool allowMove = true, bool forceAttack = false)
 		{
 			var target = Target.FromActor(targetActor);
-			if (!target.IsValidFor(Self) || target.Type == TargetType.FrozenActor)
+			if (!target.IsValidFor(Self))
 				Log.Write("lua", "{1} is an invalid target for {0}!", Self, targetActor);
 
 			if (!targetActor.Info.HasTraitInfo<FrozenUnderFogInfo>() && !Self.Owner.CanTargetActor(targetActor))
 				Log.Write("lua", "{1} is not revealed for player {0}!", Self.Owner, targetActor);
 
 			attackBase.AttackTarget(target, true, allowMove, forceAttack);
 		}
+
+		[Desc("Checks if the targeted actor is a valid target for this actor.")]
+		public bool IsValidTarget(Actor targetActor)
```

Comment:
This name is ambiguous, and used in the opposite way to the version used in the core code!

Please rename this to something like `CanTarget` to make it clearer.